### PR TITLE
[GEOT-5627] Fixed create temp file

### PR DESCRIPTION
**Pull request by ugomos**

### CODE REVIEW

File: modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java [hunk 1]
```diff
@@ -2047,7 +2047,7 @@ public void testRequestInAreaWithNoGranulesBecomesTransparent() throws Exception
         // read and check we actually got a coverage in the requested area
         GridCoverage2D coverage = reader.read(new GeneralParameterValue[] {ggp, transparent});
         assertNotNull(coverage);
-        new GeoTiffWriter(new File("/tmp/test.tiff")).write(coverage, null);
+        new GeoTiffWriter(TestData.temp(this, "test.tiff")).write(coverage, null);
```
Comment (simboss) [inline modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java]: I would add a call to dispose at the end.

**Comment by simboss** [comment]
Do we also need to backport?

Please, check if this applies to older versions.

**Comment by ugomos** [comment]
I updated this PR adding the dispose of the ImageMosaicReader.
Yes, we need also to backport, the same problem is at least in branch 16.x and 15.x.

**Comment by ugomos** [comment]
At last, the line "new GeoTiffWriter(new File("/tmp/test.tiff"))" had no reason to exist, it is now deleted. Added the dispose of the ImageMosaicReader object.

**Pull request merged**